src/vmm/src/devices/virtio/mmio.rs

```diff
@@ -186,10 +186,18 @@ impl MmioTransport {
             DRIVER_OK if self.device_status == (ACKNOWLEDGE | DRIVER | FEATURES_OK) => {
                 self.device_status = status;
                 let device_activated = self.locked_device().is_activated();
-                if !device_activated && self.are_queues_valid() {
-                    self.locked_device()
-                        .activate(self.mem.clone())
-                        .expect("Failed to activate device");
+                if !device_activated
+                    && self.are_queues_valid()
+                    && self.locked_device().activate(self.mem.clone()).is_err()
+                {
+                    self.device_status |= DEVICE_NEEDS_RESET;
```

Comment:
I think for vhost-user-block we should also set this when socket connection is closed, i.e. vhost-user-backend crashed. Maybe in later PR.

Reply:
Yeah, let's keep that for another PR, that sounds a bit more complicated :sweat_smile: